File: 118. Pascal's Triangle/solution.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
  if (numRows === 0) return [];
  if (numRows === 1) return [[1]];
  const result = [[1], [1, 1]];

  while (result.length !== numRows) {
    const currRow = [];
    const lastRow = result[result.length - 1];
    for (let i = 0; i < lastRow.length - 1; i++) {
      currRow.push(lastRow[i] + lastRow[i + 1]);
    }
    result.push([1].concat(currRow).concat([1]));
  }
  return result;
};

File: 118. Pascal's Triangle/solution.go
func generate(numRows int) [][]int {
	if numRows == 1 {
		return [][]int{{1}}
	}

	var result = [][]int{[]int{1}}

	for i := 1; i <= numRows; i++ {
		var row []int
		for j := 0; j < i; j++ {
			if j == 0 || j == i-1 {
				row = append(row, 1)
			} else {
				row = append(row, result[i-1][j-1]+result[i-1][j])
			}
		}
		result = append(result, row)
	}
	return result[1:]
}